rr_supervisor/include/supervisor.hpp

```diff
@@ -0,0 +1,57 @@
+/** @file supervisor.hpp
+ *  @brief Supervisor prototypes
+ *  @author Waleed Ahmed(w29ahmed)
+ *  @competition IARRC 2018
+ */
+
+ // Include guard
+#ifndef SUPERVISOR_H
+#define SUPERVISOR_H
+
+#include <stdio.h>
+#include <string>
+
+#include <ros/ros.h>
+#include <std_msgs/Bool.h>
+#include <std_msgs/Int8.h>
+#include <geometry_msgs/Twist.h>
+#include <std_srvs/Trigger.h>
+#include <std_srvs/Empty.h>
+
+
+class Supervisor
+{
+  public:
+    Supervisor();
+    // Callback methods
+    bool StartRace(std_srvs::Empty::Request &req, std_srvs::Empty::Response &res);
+    bool CountLap(std_srvs::Trigger::Request &req, std_srvs::Trigger::Response &res);
+    void TrackSpeed(const geometry_msgs::TwistConstPtr& msg);
+    void MonitorBattery(const std_msgs::Int8::ConstPtr& msg);
+    void FinishRace();
+  private:
+    // ROS Variables
+    ros::NodeHandle nh_;
+    ros::Publisher twist_pub;
```

Comment:
member variables should end with an underscore https://google.github.io/styleguide/cppguide.html#General_Naming_Rules